src/client/tile.cpp

```diff
@@ -1812,6 +1812,157 @@ bool TextureSource::generateImagePart(std::string part_of_name,
+						<< "for part_of_name=\"" << part_of_name
+						<< "\", cancelling." << std::endl;
+				return false;
+			}
+			driver->setRenderTarget(rt, true, true, video::SColor(0,0,0,0));
+
+			scene::IMesh *mesh = imesh->mesh;
+			driver->clearZBuffer();
+
+			core::matrix4 ProjMatrix;
+			ProjMatrix.buildProjectionMatrixOrthoLH(2.0f, 2.0f, -1.0f, 100.0f);
+
+			core::matrix4 ViewMatrix;
+			ViewMatrix.buildProjectionMatrixOrthoLH(2.0f, 2.0f, -1.0f, 100.0f);
+			ViewMatrix.setTranslation(core::vector3df(
+				1.0f * (res_rect.LowerRightCorner.X + res_rect.UpperLeftCorner.X -
+						res_rect.LowerRightCorner.X - res_rect.UpperLeftCorner.X) /
+						res_rect.getWidth(),
+				1.0f * (res_rect.LowerRightCorner.Y + res_rect.UpperLeftCorner.Y -
+						res_rect.LowerRightCorner.Y - res_rect.UpperLeftCorner.Y) /
+						res_rect.getHeight(),
+				0.0f));
+
+			driver->setTransform(video::ETS_PROJECTION, ProjMatrix);
+			driver->setTransform(video::ETS_VIEW, ViewMatrix);
+
+			core::matrix4 matrix;
+			matrix.makeIdentity();
+
+			driver->setTransform(video::ETS_WORLD, matrix);
+			driver->setViewPort(res_rect);
+
+			video::SColor basecolor = def.color;
+
+			u32 mc = mesh->getMeshBufferCount();
+			for (u32 j = 0; j < mc; ++j) {
+				scene::IMeshBuffer *buf = mesh->getMeshBuffer(j);
+				// we can modify vertices relatively fast,
+				// because these meshes are not buffered.
+				assert(buf->getHardwareMappingHint_Vertex() == scene::EHM_NEVER);
+				video::SColor c = basecolor;
+
+				if (imesh->buffer_colors.size() > j) {
+					ItemPartColor *p = &imesh->buffer_colors[j];
+					if (p->override_base)
+						c = p->color;
+				}
+
+				if (imesh->needs_shading)
+					colorizeMeshBuffer(buf, &c);
+				else
+					setMeshBufferColor(buf, c);
+
+				video::SMaterial &material = buf->getMaterial();
+				material.MaterialType = video::EMT_TRANSPARENT_ALPHA_CHANNEL_REF;
+				material.Lighting = false;
+				driver->setMaterial(material);
+				driver->drawMeshBuffer(buf);
+			}
+
+			// draw the inventory_overlay
+			if (def.type == ITEM_NODE && def.inventory_image.empty() &&
+					!def.inventory_overlay.empty()) {
+				video::ITexture *overlay_texture = getTexture(def.inventory_overlay);
+				core::dimension2d<u32> dimens = overlay_texture->getOriginalSize();
+				core::rect<s32> srcrect(0, 0, dimens.Width, dimens.Height);
+				draw2DImageFilterScaled(driver, overlay_texture, res_rect, srcrect, &res_rect, 0, true);
+			}
+
+			driver->setRenderTarget(nullptr, false, false, video::SColor());
+			driver->setTransform(video::ETS_VIEW, oldViewMat);
+			driver->setTransform(video::ETS_PROJECTION, oldProjMat);
+			driver->setTransform(video::ETS_WORLD, oldWorldMat);
+			driver->setViewPort(oldViewPort);
+
+			baseimg = driver->createImageFromData(
+				rt->getColorFormat(),
+				rt->getSize(),
+				rt->lock(),
+				false 	// copy memory
+			);
+
+			rt->unlock(); // wondering whether this is required
+			driver->removeTexture(rt);
+			// rt->drop(); isn't required
```

Comment:
dead code

Reply:
If it must not be dropped, it's a good notice to programmers. However,  `wondering whether this is required` is much worse for being unclear. Perhaps this needs a valgrind run to ensure there are no memory leaks.